opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -320,7 +318,7 @@ impl OtlpHttpClient {
     fn build_logs_export_body(
         &self,
         logs: LogBatch<'_>,
-    ) -> opentelemetry_sdk::logs::LogResult<(Vec<u8>, &'static str)> {
+    ) -> Result<(Vec<u8>, &'static str), Box<dyn std::error::Error>> {
```

Comment:
Is this the right type for this?

it looks like:
1/ It is private
2/ It is only used in `OtlpHttpClient`'s `LogExport` impl, in `export` 

It seems like it can **only** be a serde_json Error - do we lose anything if we return this?

Reply:
fixed to return serde_json::Error only

Reply:
Refactored and modified to use dyn Error itself, as the serde::json is only applicable in json feature flag.